### DISCUSSION TITLE
Starlight blog on https://teletype.in/@starlight-js

### DISCUSSION THREAD

**Discussion by playXE** [Starlight-JS/starlight#90] [Ideas]
What do you guys think about it? @andrieshiemstra @jameslahm

**Comment by andrieshiemstra** [comment]
Hi, I don't know teletype.in specifically but i like the idea to blog about starlight, and js engines in general

The next two weeks i'm in itally (with only a crappy old linux laptop) and I was planning to do some writing about my plans for my JS-Engine abstraction layer. 

What are you planning to write about? you guys seem a bit more into the nitty gritty low level stuff which i know a little less about. Looking forward to learning about your ideas on that!

**Reply by playXE** [reply]
I would like to write about the internals of JS engines and maybe some guides on how to implement [insert name there] for starters. :) 

All the popular JS engines have fancy blogs but usually they explain internals briefly without much details and when I was started to develop Starlight these blogs helped me just a little but now when I have JS engine I can finally write better blog about internals. :D

**Reply by andrieshiemstra** [reply]
> All the popular JS engines have fancy blogs but usually they explain internals briefly without much details and when I was started to develop Starlight these blogs helped me just a little but now when I have JS engine I can finally write better blog about internals. :D

It would be a nice change for JS-Engines, from my experience with quickjs and spidermonkey i do feel the documentation was.. well.. sub optimal.

I also feel a good public API with up-to-date rustdocs can do wonders to encourage adoptation of your project.  let's talk later about how we see the public API. And already looking forward to reading your blogs!

I gotta go pack the car now :) i'll be online every now and then the next two weeks, after that i plan to go full throttle on Promises/async!

Btw, how did your tests go?

**Reply by playXE** [reply]
Tests went nicely. 

> after that i plan to go full throttle on Promises/async!


That would be awesome! I hope we could get working async and promises by the end of the summer :)

**Reply by jameslahm** [reply]
I also don't use teletype.in before. But I just see the two blogs. And I think it's great!
